### ISSUE TITLE
Constructing Graphframe from Dataframes - 'id', 'src' and 'dst' column name requirements - allow user defined column names

### ISSUE THREAD

**Issue by m316257** [graphframes/graphframes#241]
When I construct a Graphframe from a vertex Dataframe and an edge Dataframe, the vertex Dataframe needs a column called 'id' that contains the ids for the vertices and the edge Dataframe has a similar requirement for the 'src' and 'dst' columns.

If my API call naturally returns a Dataframe where the column containing vertex ids has a different name, for example [here](https://gchq.github.io/gaffer-doc/getting-started/spark-operation-examples.html#getdataframeofelements-example) where it's called 'vertex', I need to rename the column to 'id' before constructing the Graphframe. In this case renaming the column adds an unnecessary overhead.

Allowing a user to specify the names of the columns containing the vertex 'ids' and 'src' and 'dst' in the Dataframes to be used could enhance performance.

**Comment by felixcheung** [comment]
This is a good point, I think it'd better if the column is configurable.
Although, renaming a column in Spark should be really light-weighted?

**Comment by SemyonSinchenko** [comment]
I would like to handle complex problems like custom ids in the same way, like property-graphs. I'm closing this one in favor of #565

**Comment by rjurney** [comment]
I don’t think we should accommodate this feature. It is helpful to be able to know these column names throughout the codebase.

**Comment by SemyonSinchenko** [comment]
> I don’t think we should accommodate this feature. It is helpful to be able to know these column names throughout the codebase.

If we have a property graph abstraction, we can easily map custom ids to the underlying GraphFrame with constant ids names. PropertyGraph in that case is just a thin layer on top of existing GraphFrame
